PVTPackage/source/MultiphaseSystem/PhaseModel/CubicEOS/CubicEoSPhaseModel.cpp

```diff
@@ -107,6 +107,16 @@ double CubicEoSPhaseModel::computeCompressibilityFactor( double pressure,
   }
   else
   {
+
+    // Check for unphysical roots and remove them 
+    for( std::size_t i=0; i< sols.size(); ++i)
+    {
+      if(sols[i] <= mixCoeffs.BMixture)
+      {
+        sols.erase(sols.begin()+i);
+      }
+    }
```

Comment:
Iterating on `sols` while you are modifying the size of `sols` does not sound good.
As a matter of fact
```c++
std::vector<int> sols{-1, -2, -3, -4};
for (std::size_t i = 0; i < sols.size(); ++i) {
    if (sols[i] < 0) {
        sols.erase(sols.begin() + i);
    }
}
for (const int & s: sols) std::cerr << s << " ";
```
prints `-2 -4`. This loop improperly deals with two consecutive values to be removed.
You should prefer `copy_if` or store the indices first and `erase` as a second step.

Reply:
Or use the remove-erase idiom:
```c++
auto const unphysical = [&]( double v ) { return v <= mixCoeffs.Bmixture; };
sols.erase( std::remove_if( sols.begin(), sols.end(), unphysical ), sols.end() );
```

Reply:
Well, that was embarassing.  I pushed a fix.